cpp/src/runtime/context.cpp

```diff
@@ -25,10 +26,16 @@
 
 namespace cudf {
 
-context::context() : _program_cache{std::make_unique<jit::program_cache>()} {}
+context::context() : _program_cache{std::make_unique<jit::program_cache>()}
+{
+  auto dump_codegen_flag = getenv_or("LIBCUDF_DUMP_CODEGEN", std::string{"OFF"});
```

Comment:
Nit: adding `JIT` prefix make the env parameter easier to spot.
```suggestion
  auto dump_codegen_flag = getenv_or("LIBCUDF_JNIT_DUMP_CODEGEN", std::string{"OFF"});
```